zerver/lib/message.py

```diff
@@ -227,7 +228,10 @@ def message_to_dict_json(message: Message, realm_id: Optional[int] = None) -> by
 
 
 def save_message_rendered_content(message: Message, content: str) -> str:
-    rendered_content = render_markdown(message, content, realm=message.get_realm())
+    rendering_result = render_markdown(message, content, realm=message.get_realm())
+    rendered_content = None
+    if rendering_result is not None:
+        rendered_content = rendering_result["rendered_content"]
```

Comment:
Under what circumstance might `rendering_result` be None?  It looks like `render_markdown` never returns None.

Reply:
There is a strange test that patches `render_markdown` to return `None`,

Reply:
In `test_message_dict`, we have
```python
    @mock.patch("zerver.lib.message.markdown_convert")
    def test_applying_markdown_invalid_format(self, convert_mock: Any) -> None:
        # pretend the converter returned an invalid message without raising an exception
        convert_mock.return_value = None
        sender = self.example_user("othello")
        receiver = self.example_user("hamlet")
        recipient = Recipient.objects.get(type_id=receiver.id, type=Recipient.PERSONAL)
        sending_client = make_client(name="test suite")
        message = Message(
            sender=sender,
            recipient=recipient,
            content="hello **world**",
            date_sent=timezone_now(),
            sending_client=sending_client,
            last_edit_time=timezone_now(),
            edit_history="[]",
        )
        message.set_topic_name("whatever")
        message.save()

        # An important part of this test is to get the message through this exact code path,
        # because there is an ugly hack we need to cover.  So don't just say "row = message".
        row = MessageDict.get_raw_db_rows([message.id])[0]
        dct = MessageDict.build_dict_from_raw_db_row(row)
        error_content = (
            "<p>[Zulip note: Sorry, we could not understand the formatting of your message]</p>"
        )
        self.assertEqual(dct["rendered_content"], error_content)
```
which might be a legacy

Reply:
I see; there may be some follow-up to try to delete that code, but this seems fine to test that error-handling code path.